lib/parsers/asm-parser-spirv.js

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022, Compiler Explorer Authors
```

Comment:
Could you port this file over to typescript? We are trying to use it in more places and new files are easier to change. It should be as simple as renaming the file and adding a few types

Reply:
It sounds like an easy task... but I have no experience with js nor ts.
I have renamed the file to `.ts` and made those changes:
```
modified   lib/parsers/asm-parser-spirv.ts
@@ -44,10 +44,10 @@ export class SPIRVAsmParser extends AsmParser {
         return files;
     }

-    processAsm(asmResult, filters) {
+    override processAsm(asmResult, filters) {
         const startTime = process.hrtime.bigint();

-        const asm = [];
+        const asm: any = [];

         let asmLines = utils.splitLines(asmResult);
         const startingLineCount = asmLines.length;
@@ -61,7 +61,7 @@ export class SPIRVAsmParser extends AsmParser {
         const comment = /;/;
         const opLine = /OpLine/;
         const opExtDbg = /OpExtInst\s+%void\s+%\d+\s+Debug/;
-        let source = null;
+        let source: any = null;

         for (let line of asmLines) {
             let match = line.match(sourceTag);
```
Everything compiles well, but I have lost the color highlighting, so something does not work as before obviously.
Any ideas on what I am doing wrong, or what I could try?